### PR TITLE
MINIFI-1199: Integrate H2O Driverless AI PSP in MiNiFi

### PR DESCRIPTION
**Pull request by james94**
**MiNiFi C++ and H2O Driverless AI Integration** via Custom Python Processors:

Integrates MiNiFi C++ with H2O's Driverless AI by Using Driverless AI's Python Scoring Pipeline (PSP) and MiNiFi's Custom Python Processors. Uses the Python Processors to execute the Python Scoring Pipeline scorer to do batch scoring and real-time scoring for one or more predicted labels on test data in the incoming flow file content. I would like to contribute my processors to MiNiFi C++ as a new feature.

**3 custom python processors** created for MiNiFi:

**H2oPspScoreRealTime** - Executes H2O Driverless AI's Python Scoring Pipeline to do interactive scoring (real-time) scoring on an individual row or list of test data within each incoming flow file.

**H2oPspScoreBatches** - Executes H2O Driverless AI's Python Scoring Pipeline to do batch scoring on a frame of data within each incoming flow file.

**ConvertDsToCsv** - Converts data source of incoming flow file to csv. 

**3 Follow Up Questions** for future improvement:

I do have a question about Python Processors, I notice the ExecutePythonProcessor has a Module Directory property for passing in paths to files and/or directories which contain modules required by the script. I am wondering what if there is a "module name" that the user needs to enter in the processor property to update the scripts import "module name" before that line of code is executed? Is that possible? and if so, how would one do that?

We do require the user to set Driverless AI License Key as an **environment variable**, so they can run the python scoring pipeline. I am aware I can add a property for the License Key for the python processors **H2oPspScoreRealTime**, **H2oPspScoreBatches**, but is there a way once those properties are set, that inside the processor we can set the environment variable temporarily? but then again, we would have to set the same License Key for each processor that uses the python scoring pipeline. Would a controller service help in accomplishing this goal?

Is there a place on the repo I can add config-batch-scoring.yml and config-interactive-scoring.yml files to show users how to get started with these processors quickly?

Created Jira ticket associated with this PR: https://issues.apache.org/jira/browse/MINIFICPP-1199

- Note: more information about the 3 python processors is available in the Jira link above.

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by phrocker** [comment]
Thanks @james94  I'll take a look.

**Comment by james94** [comment]
@phrocker I looked at the travis and appveyor failures and they look unrelated to my PR. I didn't see any errors relating to the 3 python processors above.

File: extensions/pythonprocessors/h2o/dai/psp/H2oPspScoreBatches.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        # read flow file tabular data content into read_cb.content data member
+        session.read(flow_file, read_cb)
+        # instantiate H2O's python scoring pipeline scorer
+        scorer = Scorer()
+        # load tabular data str into datatable
+        test_dt_frame = dt.Frame(read_cb.content)
+        # do batch scoring on test data in datatable frame, return pandas df with predicted labels
+        batch_scores_df = scorer.score_batch(test_dt_frame)
+        # convert df to str, then write to flow file
+        batch_scores_df_str = batch_scores_df.to_string()
```
Comment (james94) [inline extensions/pythonprocessors/h2o/dai/psp/H2oPspScoreBatches.py:87]: @phrocker I want to make one update to the **H2oPspScoreBatches.py** processor.

I noticed the batch predictions are stored into a csv file with an extra column called index. Index starts from 0 to number of rows - 1. I realized after the **H2oPspScoreBatches** processor makes batch predictions, it inserts an index column for the first column and the second column are the predictions. The index column is not needed and can be removed by adding parameter **index=False** to the **batch_scores_df.to_string()** function on line 87.

**Comment by james94** [comment]
@phrocker I moved the **ConvertDsToCsv** python processor to the `h2o/` base directory since it is not only meant to be used with the 2 Python Scoring Pipeline processor(s): **H2oPspScoreRealTime** and **H2oPspScoreBatches**, but it can also be used with the 1 MOJO Scoring Pipeline processor: **H2oMojoPwScoring**. **ConvertDsToCsv** can also be used without the scoring pipeline processors just to convert a variety of data sources to csv.

Note: I do have another PR for **H2oMojoPwScoring** python processor: [MINIFI-1201: Integrate H2O Driverless AI MSP in MiNFi](https://github.com/apache/nifi-minifi-cpp/pull/766).

 I looked at the travis failures and they look unrelated to this update in the PR.

**Pull request merged**